components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -221,6 +224,29 @@ func InjectOAuthProxy(notebook *nbv1.Notebook, oauth OAuthConfig) error {
 	return nil
 }
 
+func (w *NotebookWebhook) ClusterWideProxyIsEnabled() bool {
```

Comment:
```suggestion
func (w *NotebookWebhook) ClusterWideProxyIsEnabled() (map[string]string, bool)  {
```

I think that `proxyEnvVars` does not need to be a global variable, especially since every call to `ClusterWideProxyIsEnabled` creates its content anew. Can you return the proxy env vars from the function?

Reply:
Hi @jiridanek very good point in general. There are certainly improvements possible, I am unsure whether now (adding back previous functionality) or later. I or you could create a new ticket for tracking with your suggestion and link it to issue #291